src/common/src/util/sort_util.rs

```diff
@@ -62,55 +64,171 @@ pub enum NullsAre {
     Smallest,
 }
 
+impl NullsAre {
+    pub fn from_protobuf(nulls_are: &PbNullsAre) -> Self {
+        match nulls_are {
+            PbNullsAre::Largest => Self::Largest,
+            PbNullsAre::Smallest => Self::Smallest,
+            PbNullsAre::Unspecified => unreachable!(),
+        }
+    }
+
+    pub fn to_protobuf(self) -> PbNullsAre {
+        match self {
+            Self::Largest => PbNullsAre::Largest,
+            Self::Smallest => PbNullsAre::Smallest,
+        }
+    }
+}
+
+/// Order type of a column.
 #[derive(PartialEq, Eq, Hash, Copy, Clone, Debug, Default)]
 pub struct OrderType {
     direction: Direction,
-    // TODO(rc): enable `NULLS FIRST | LAST`
-    // nulls_are: NullsAre,
+    nulls_are: NullsAre,
 }
 
 impl OrderType {
     pub fn from_protobuf(order_type: &PbOrderType) -> OrderType {
         OrderType {
             direction: Direction::from_protobuf(&order_type.direction()),
+            nulls_are: NullsAre::from_protobuf(&order_type.nulls_are()),
         }
     }
 
     pub fn to_protobuf(self) -> PbOrderType {
         PbOrderType {
             direction: self.direction.to_protobuf() as _,
+            nulls_are: self.nulls_are.to_protobuf() as _,
         }
     }
 }
 
 impl OrderType {
-    pub const fn new(direction: Direction) -> Self {
-        Self { direction }
+    pub fn new(direction: Direction, nulls_are: NullsAre) -> Self {
+        Self {
+            direction,
+            nulls_are,
+        }
     }
 
-    /// Create an ascending order type, with other options set to default.
-    pub const fn ascending() -> Self {
+    pub fn from_bools(asc: Option<bool>, nulls_first: Option<bool>) -> Self {
+        let direction = match asc {
+            None => Direction::default(),
+            Some(true) => Direction::Ascending,
+            Some(false) => Direction::Descending,
+        };
+        match nulls_first {
+            None => Self::new(direction, NullsAre::default()),
+            Some(true) => Self::nulls_first(direction),
+            Some(false) => Self::nulls_last(direction),
+        }
+    }
+
+    pub fn ascending(nulls_are: NullsAre) -> Self {
         Self {
             direction: Direction::Ascending,
+            nulls_are,
         }
     }
 
-    /// Create an descending order type, with other options set to default.
-    pub const fn descending() -> Self {
+    pub fn descending(nulls_are: NullsAre) -> Self {
         Self {
```

Comment:
A nit: can we make private all of the following methods to avoid confusion and misuse? I think `from_bools` and `is_ascending`, `nulls_are_smallest` are enough.

Reply:
Maybe we don't even need these helper constructors, jsut construct directly 🤪. And also boolean functions, just make public the fields since they are simple enums ("data types")

Reply:
Good idea to make these ctors private. But I don't want developers directly use `Direction` and `NullsAre` in other crates, that can lead to misuse, especially `Direction`. Let me also try make these two enum private.